src/request.ts

```diff
@@ -133,7 +140,14 @@ const _fetch = (options: RequestOptions) => {
     fetch!(url, {
         method: options?.method || 'GET',
         headers,
-        keepalive: options.method === 'POST',
+        // if body is greater than 64kb, then fetch with keepalive will error
+        // see 8:10:5 at https://fetch.spec.whatwg.org/#http-network-or-cache-fetch,
+        // but we do want to set keepalive sometimes as it can  help with success
+        // when e.g. a page is being closed
+        // so let's get the best of both worlds and only set keepalive for POST requests
+        // where the body is less than 64kb
+        // NB this is fetch keepalive and not http keepalive
+        keepalive: options.method === 'POST' && (estimatedSize || 0) < 64 * 1024,
```

Comment:
Could abstract this but doesn't seem necessary given the comment explains a lot
```suggestion
        keepalive: options.method === 'POST' && (estimatedSize || 0) < MAX_KEEPALIVE_KB,
```

Reply:
yeah, i'm always torn by the tension between "clarify intent" and "have fewest parts"
if you make it a const it's arguably clearer but i need to look at multiple places to be sure i understand